### DISCUSSION TITLE
Including "counts" dataset in conversion to AnnData

### DISCUSSION THREAD

**Discussion by roussine** [satijalab/seurat#8769] [Q&A]
Hello all -

as I can see, the Convert function of SeuratDisk (from h5Seurat to h5ad) will never touch the counts dataset (not-normalised matrix) of the default assay (e.g., "RNA") in a h5Seurat object, as it gives priority to "scale.data" (variable features only) and then "data" (normalised matrix) when filling X.

Is there a way to force such conversion?.. Or maybe use an alternative to SeuratDisk (as no longer supported..)? sceasy seems not a solution, as it picks only a normalised matrix and ignores the rest.

Thanks a lot in advance for any help
Leo

**Comment by ccnawrocki** [comment]
I see this is from awhile ago, but I just did it by doing this in R: 
```
obj@assays$RNA@data <- obj@assays$RNA@counts
SaveH5Seurat(object = obj, filename = "obj_rawcts.H5Seurat", overwrite = T) # Using SeuratDisk package
Convert(source = "obj_rawcts.H5Seurat", dest = "obj_rawcts.h5ad", overwrite = T) # Using SeuratDisk package
```

Then in Python: 
```
obj = sc.read_h5ad("obj_rawcts.h5ad")
```

Now, the `anndata` object has the counts in `obj.raw.X`. This will be `float`, so you'll want to convert to `integer` before doing anything else. You can just reproduce the normalized data by re-doing the normalization step with `scanpy` or by doing it manually (pretty feasible, if you had done log norm in `Seurat`).

Maybe this will help someone else down the line.
